src/utils/contracts.ts

```diff
@@ -108,10 +132,22 @@ export const getFlashMintLeveragedContractForToken = (
   signerOrProvider: Signer | Provider | undefined,
   chainId: number = ChainId.Polygon
 ): Contract => {
+  if (
+    chainId === ChainId.Arbitrum &&
+    (token === IndexCoopBitcoin2xIndex.symbol ||
+      token === IndexCoopEthereum2xIndex.symbol)
+  ) {
+    return getIndexFlashMintLeveragedExtendedContract(signerOrProvider)
+  }
   switch (token) {
     case BTC2xFlexibleLeverageIndex.symbol:
     case ETH2xFlexibleLeverageIndex.symbol:
       return getFlashMintLeveragedForCompoundContract(signerOrProvider)
+    case IndexCoopBitcoin3xIndex.symbol:
+    case IndexCoopEthereum3xIndex.symbol:
+    case IndexCoopInverseBitcoinIndex.symbol:
+    case IndexCoopInverseEthereumIndex.symbol:
+      return getIndexFlashMintLeveragedExtendedContract(signerOrProvider)
```

Comment:
does it make sense to have 2 separate switch statements here (1 for mainnet and 1 for arbitrum), or trying to change these as little as possible for now?

Reply:
That's right, that's probably cleaner 👍 Updated.